src/test/java/com/rybka/repository/CurrencyHistoryRepositoryTest.java

```diff
@@ -0,0 +1,122 @@
+package com.rybka.repository;
+
+import com.rybka.model.CurrencyHistory;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@ExtendWith(SpringExtension.class)
+@DataJpaTest
+class CurrencyHistoryRepositoryTest {
+    @Autowired
+    private CurrencyHistoryRepository repository;
+
+    @Test
+    public void testOnCorrectHistorySave() {
+
+        // given
+        var currencyBase = "USD";
+        var currencyTarget = "EUR";
+        var currencyRate = 0.88;
+        var currencyAmount = 5d;
+        var currencyTotal = currencyRate * currencyAmount;
+        var testData = new CurrencyHistory(currencyBase, currencyTarget, currencyAmount, currencyRate, currencyTotal);
+
+        // when
+        var savedData = repository.save(testData);
+
+        // then
+        assertEquals(testData.getBase(), savedData.getBase());
```

Comment:
What about other values. Are you sure they returned from db?

Reply:
fixed